database/schema/controller.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package schema
+
+import "strings"
+
+// ControllerDDL is used to create the controller database schema at bootstrap.
+func ControllerDDL() []string {
+	delta := `
+CREATE TABLE lease_type (
+	id		INT PRIMARY KEY,
+	type	TEXT
+);
+
+INSERT INTO lease_type VALUES
+	(0, 'controller'), 
+	(1, 'model' ), 
+	(2, 'application');
+
+CREATE TABLE IF NOT EXISTS lease (
+	uuid			TEXT PRIMARY KEY,
+	lease_type_id	INT NOT NULL,
+	name			TEXT,
+	holder			TEXT,
+	start			TIMESTAMP,
+	duration		INT,
```

Comment:
In my head, I would have thought we'd want an 'expires: TIMESTAMP' field and then have an index on that for quick lookup, rather than storing that in memory the way we currently do.
But I guess maybe we want expiry to be hard cached just in memory rather than churning in the DB?

Reply:
We could schedule the next check as we currently do and still use such a column. I've made the change.